### PR TITLE
Add merge_group webhook event via MergeGroupEvent and MergeGroup structs

### PR DESCRIPTION
**Pull request by lyoung-confluent**
Adds the structs for the new [merge_group](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#merge_group) webhook event used by the [Pull Request Merge Queue Limited Beta](https://github.blog/changelog/2021-10-27-pull-request-merge-queue-limited-beta/). 

I'm not sure what the policy is on merging structs that are subject to change:
> Note: The pull request merge queue feature is currently in limited public beta and subject to change.

### CODE REVIEW

**Comment by gmlewis** [comment]
> I'm not sure what the policy is on merging structs that are subject to change:

We tend to live on the cutting edge in this repo, so we have no problem with that, as the GitHub v3 API seems to be a continuously-moving target in my experience. That's not a criticism... they seem to be continuously innovating, and we're just trying our best to keep up.  :smile:

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:576]:
```suggestion
// MergeGroupEvent represents activity related to merge groups in a merge queue. The type of activity is specified
```

**Review by gmlewis — APPROVED** [review]
Thank you, @lyoung-confluent !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merged.

**Pull request merged**
